apps/web-giddh/src/app/vouchers/preview/preview.component.ts

```diff
@@ -1,90 +1,1170 @@
+     * Handle upload file
+     *
+     * @memberof VouchersPreviewComponent
+     */
+    public uploadFile(): void {
+        const selectedFile: any = document.getElementById("csv-upload-input");
+        if (selectedFile?.files?.length) {
+            const file = selectedFile?.files[0];
+
+            this.generalService.getSelectedFile(file, (blob, file) => {
+                this.isFileUploading = true;
+
+                // =========== move in component store ========== //
+                this.componentStore.uploadFile({ postRequestObject: { file: blob, fileName: file.name } });
+                //         this.salesService.updateAttachmentInVoucher(requestObject).pipe(takeUntil(this.destroyed$)).subscribe((res) => {
+                //             if (res.status === "error") {
+                //                 this.toaster.showSnackBar('error', res?.message);
+                //             }
+
+                //             this.downloadVoucherPdf('base64');
+                //         }, () => this.toaster.errorToast(this.commonLocaleData?.app_something_went_wrong));
+
+            });
+        }
+    }
+
+    /**
+     * Download Invoice PDF
+     *
+     * @return {*}  {void}
+     * @memberof VouchersPreviewComponent
+     */
+    public downloadPdf(): void {
+        if (this.isVoucherDownloading || this.isVoucherDownloadError) {
+            return;
+        }
+
+        if ([VoucherTypeEnum.estimate, VoucherTypeEnum.generateEstimate, VoucherTypeEnum.proforma, VoucherTypeEnum.generateProforma].includes(this.voucherType)) {
+            if (this.selectedInvoice && this.selectedInvoice.blob) {
+                return saveAs(this.selectedInvoice.blob, `${this.selectedInvoice.account.name} - ${this.selectedInvoice.voucherNumber}.pdf`);
+            } else {
+                return;
+            }
+        } else if (this.voucherType === VoucherTypeEnum.creditNote || this.voucherType === VoucherTypeEnum.debitNote) {
+            if (this.selectedInvoice?.hasAttachment) {
+                // this.downloadVoucherModal?.show();
+                // "download-voucher" this componnent is missing in design
+            } else {
+                if (this.selectedInvoice) {
+                    return saveAs(this.selectedInvoice.blob, `${this.selectedInvoice.voucherNumber}.pdf`);
+                }
+            }
+        } else if (this.voucherType === VoucherTypeEnum.purchase) {
+            if (this.pdfPreviewHasError || !this.pdfPreviewLoaded) {
+                return;
+            }
+            if (!this.selectedInvoice?.hasAttachment) {
+                let voucherNumber = (this.selectedInvoice?.voucherNumber) ? this.selectedInvoice?.voucherNumber : this.commonLocaleData?.app_not_available;
+                saveAs(this.attachedDocumentBlob, voucherNumber + '.pdf');
+            } else {
+                // this.downloadVoucherModal?.show();
+                // "download-voucher" this componnent is missing in design
+            }
```

Comment:
// "download-voucher" this componnent is missing in design??????????

Reply:
this dialog design in material was pending that time now it is fixed please check